x-pack/plugins/observability_solution/observability_shared/public/hooks/use_breadcrumbs.ts

```diff
@@ -33,23 +43,49 @@ function addClickHandlers(
 }
 
 function getTitleFromBreadCrumbs(breadcrumbs: ChromeBreadcrumb[]) {
-  return breadcrumbs.map(({ text }) => text?.toString() ?? '').reverse();
+  return breadcrumbs
+    .map(({ text }) => text?.toString() ?? '')
+    .reverse()
+    .concat([OBSERVABILITY_TEXT]);
 }
 
+/**
+ *
+ * By default the breadcrumbs will be passed to either serverless.setBreadcrumbs or chrome.setBreadcrumbs depending on the
+ * environment. The breadcrumbs will *also* be passed to the project style breadcrumbs for stateful project style. We will use "project style"
+ * here to refer to serverless chrome and stateful project style chrome. Classic refers to stateful classic chrome.
+ *
+ * Project style breadcrumbs add a root crumb ("deployment" etc) and "nav crumbs" which are derived from the navigation structure. By default
+ * the "absolute" mode is used which means the breadcrumbs passed here will omit the navigation derived "nav crumbs". You can pass
+ * absoluteProjectStyleBreadcrumbs: false to include the 'smart' "nav crumbs".
+ *
+ * In classic mode (not project style) the 'Observability' crumb is added.
+ *
+ * You can also pass classicOnly to only set breadrumbs in the classic chrome context. This can be useful if your solution just wants to defer all project style crumbs to the "nav crumbs".
+ */
 export const useBreadcrumbs = (
   extraCrumbs: ChromeBreadcrumb[],
   options?: {
     app?: { id: string; label: string };
     breadcrumbsAppendExtension?: ChromeBreadcrumbsAppendExtension;
     serverless?: ServerlessPluginStart;
+    absoluteProjectStyleBreadcrumbs?: boolean;
+    classicOnly?: boolean;
   }
 ) => {
   const params = useQueryParams();
   const { app, breadcrumbsAppendExtension, serverless } = options ?? {};
+  const absolute = options?.absoluteProjectStyleBreadcrumbs === false ? false : true;
```

Comment:
I've tried to use the default that will cause the least surprises. With absolute if a solution passes in a full set of crumbs (including the application name etc) it will look consistent across the 3 modes, it might not be perfect but it will absolutely make sense. Whereas with `false` the nav crumbs come into play and the results are a lot more surprising if the solution hasn't taken that into account.

`classicOnly` is really just there to bypass everything project based.